hw completed

maksar24 opened the pull request.

maksar24 committed:
hw completed

KirillKulakow commented on .env:
```diff
@@ -0,0 +1 @@
+DB_HOST=mongodb+srv://maksar24:2eQeGkCrBQYpkWrh@cluster0.u5p3g.mongodb.net/db-contacts?retryWrites=true&w=majority
```
удалить с гита и добавить файл в gitignore

maksar24 replied:
Окей....Я просто не добавил его изначально в .gitignore, чтобы можно было моей базой воспользоваться при проверке

KirillKulakow replied:
не переживай - если что я на своей потестирую

KirillKulakow commented on controllers/contacts/updateContact.js:
```diff
@@ -1,10 +1,10 @@
 const updateContact = async (req, res, next) => {
   const { contactId } = req.params
-  const contact = await contactsOperations.updateContact(contactId, req.body)
+  const contact = await Contact.findByIdAndUpdate(contactId, req.body, { new: true })
```
https://mongoosejs.com/docs/api.html#model_Model.findByIdAndUpdate
не вижу там такой опции как new, лучше использовать returnDocument

maksar24 replied:
это преподаватель просто на занятии показал так

maksar24 replied:
new: bool - if true, return the modified document rather than the original. defaults to false (changed in 4.0)

maksar24 replied:
можно пожалуйста оставить { new: true }, если я нашел это в доках?...

KirillKulakow replied:
ок, там есть промежуточный обработчик при помощи которого можно использовать данную опцию

KirillKulakow commented on controllers/contacts/updateStatusContact.js:
```diff
@@ -0,0 +1,22 @@
+const updateStatusContact = async (req, res, next) => {
+  const { contactId } = req.params
+  const { favorite } = req.body
+  const contact = await Contact.findByIdAndUpdate(contactId, { favorite }, { new: true })
```
так же

KirillKulakow reviewed (changes requested):
Need fix

maksar24 committed:
fix:
1. added new validation in updateContact and updateStatusContact;
2. added .env to file .gitignore

maksar24 committed:
moved this file to .gitignore
